### ISSUE TITLE
about V

### ISSUE THREAD

**Issue by INTOUCHABLE-VS** [hkchengrex/STCN#129]
When generating V,  you cat the image and mask. This is different from STM.  What will be the impact on the results?

**Comment by INTOUCHABLE-VS** [comment]
STCN：f = torch.cat([image, mask], 1) in ValueEncoderSO to product v.
STM: x = self.conv1(f) + self.conv1_m(m) + self.conv1_o(o) in Encoder_m to product the k and v for memery bank.
What will be the impact on the results?

**Comment by hkchengrex** [comment]
They are the same.

Proof:
```python
import torch
import torch.nn as nn

img = torch.randn((2, 3, 64, 64))
msk = torch.randn((2, 1, 64, 64))

cat_conv = nn.Conv2d(4, 64, kernel_size=3, padding=1, bias=False)
add_conv_i = nn.Conv2d(3, 64, kernel_size=3, padding=1, bias=False)
add_conv_m = nn.Conv2d(1, 64, kernel_size=3, padding=1, bias=False)

add_conv_i.weight.data = cat_conv.weight.data[:, :3]
add_conv_m.weight.data = cat_conv.weight.data[:, 3:]

cat_out = cat_conv(torch.cat([img, msk], 1))
add_out = add_conv_i(img) + add_conv_m(msk)

print(torch.allclose(cat_out, add_out, atol=1e-6))

```

**Comment by INTOUCHABLE-VS** [comment]
I understand. Thanks for your reply
